lib/services/api/api_exception.dart

```diff
@@ -25,6 +25,11 @@ class ApiException implements Exception {
     );
   }
 
+  static const wrongResponseStructure = ApiException(
```

Comment:
naming should come with the class name unless they are primitive type or a simple field. In this case, I believe you're creating a `const` a.k.a static field so it should be sth like `wrongResponseException`.

btw, the name `wrong response` is kinda broad 🤔  if it's a default or unclear exception you may want to consider `IllegalResponseException` or `InvalidResponseStructureException` ?

Reply:
`InvalidResponseStructureException` is perfect excepts `Exception` suffix 😁 

I still prefer to see it like this 
```dart
ApiException.invalidResponseStructure
```
rather than 
```dart
ApiException.invalidResponseStructureException
```
 🤔

Reply:
@markgravity why don't you create `InvalidResponseStructureException` class and extends `ApiException`? 😅 or `factory ApiException.invalidResponseStructure()`

Reply:
@manh-t I want to trick this class as an enum so that I follow the way the flutter team defines an enum class 😁 
```dart
/// The thickness of the glyphs used to draw the text
class FontWeight {
  const FontWeight._(this.index);

  /// The encoded integer value of this font weight.
  final int index;

  /// Thin, the least thick
  static const FontWeight w100 = FontWeight._(0);

  /// Extra-light
  static const FontWeight w200 = FontWeight._(1);

  /// Light
  static const FontWeight w300 = FontWeight._(2);
 ```

Reply:
@markgravity okay, shall you make the constructor private? I see that you do not create public constructor anywhere, and I believe that if you treat this class as enum, you will not create class instance outside, am I right? 😄

Reply:
@manh-t I think it likes create enum case from raw values.
 So if we have an error response from server, then
```dart
const invalidToken = ApiException(
 source: "unauthorized",
 message: "The access token is invalid",
 code: "invalid_token",
); // the input from api response

ApiException.invalidToken == invalidToken; // true
```
I updated it with equatable in this [branch](https://github.com/markgravity/flutter-survey/blob/bug/77-fix-token-doesnt-refresh-after-expires/lib/services/api/api_exception.dart)